### PR TITLE
Fix "The Main Function" example

### PR DESCRIPTION
**Pull request by riyafa**
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: examples/the-main-function/the_main_function.bal [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
 // parameter with the default value `18`. The third parameter `year` is also a defaultable parameter.
-// The rest parameter `modules` represents the additional arguments.
+// The array parameter `modules` represents the additional arguments.
+public type Student record {|
+    string name;
+    int age = 18;
+    string year =  "Freshman";
+    string[] modules;
+|};
+// The `main` function that accepts student information as options and prints out a formatted string.
```
Comment (warunalakshitha) [inline examples/the-main-function/the_main_function.bal:12]:
```suggestion
// The `main` function accepts student information as options and prints out a formatted string.
```

File: examples/the-main-function/the_main_function.description [hunk 2]
```diff
@@ -1,4 +1,10 @@
 // A `public` function named `main` is considered as an entry point to a Ballerina program.
-// The `main` function is data-binding and can have zero or more parameters whose types are subtypes of `anydata`,
+// The `main` function is data-binding and can have zero or more parameters whose types are
+// string, int, float, decimal or union of one of these types and nil.
 // including any number of required/defaultable parameters and/or a single rest parameter.
-// The `main` function could also return a value whose type is a subtype of `error?`.
+// The main parameters except for the last parameters if it is a *record, represents operands 
+// in the commandline. Order of operands are significant.
+// The last parameter can be a *record that reprents options in the command line.
```
Comment (warunalakshitha) [inline examples/the-main-function/the_main_function.description:7]:
```suggestion
// The last parameter can be a *record that represents options in the command line.
```

File: examples/the-main-function/the_main_function.description [hunk 3]
```diff
@@ -1,4 +1,10 @@
 // A `public` function named `main` is considered as an entry point to a Ballerina program.
-// The `main` function is data-binding and can have zero or more parameters whose types are subtypes of `anydata`,
+// The `main` function is data-binding and can have zero or more parameters whose types are
+// string, int, float, decimal or union of one of these types and nil.
 // including any number of required/defaultable parameters and/or a single rest parameter.
-// The `main` function could also return a value whose type is a subtype of `error?`.
+// The main parameters except for the last parameters if it is a *record, represents operands 
+// in the commandline. Order of operands are significant.
```
Comment (warunalakshitha) [inline examples/the-main-function/the_main_function.description:6]:
```suggestion
// in the command-line. The order of operands is significant.
```

File: examples/the-main-function/the_main_function.description [hunk 4]
```diff
@@ -1,4 +1,10 @@
...
+// The last parameter can be a *record that reprents options in the command line.
+// The *record field types are same as that of the parameter types. 
+// The order of option commandline arguments are insignificant.
```
Comment (warunalakshitha) [inline examples/the-main-function/the_main_function.description:9]:
```suggestion
// The order of option command-line arguments is insignificant. 
```

File: examples/the-main-function/the_main_function.out [hunk 5]
```diff
@@ -2,30 +2,31 @@
 # `.bal` file, and execute the `bal run` command below.
 
 # Use the ballerina `run` command to invoke the `main` function specifying `Alice`
-# as the string argument for `name`. `18` would be set as the value for
-# `age` and `Freshman` would be set as the value for `year`.
-bal run the_main_function.bal Alice
+# as the command line string argument for `name`. `18` would be set as the value 
+# for `age` and `Freshman` would be set as the value for `year`.
+bal run the_main_function.bal -- --name=Alice
```
Comment (MaryamZi) [inline examples/the-main-function/the_main_function.out:7]: IMO, it is better to stick to positional args (vs named args) for this feature.

If it is only the rest param that is not working atm, IMO we can remove just that bit from the example and keep the rest as is. And once the rest param issue is fixed we can add back the content.

Comment (riyafa) [reply examples/the-main-function/the_main_function.out:7]: We changed it to use options for now. We plan to write a proper example when the PR fixing operand processing is merged for release. There has been a complete rewrite of the command line argument processing logic: https://docs.google.com/document/d/1Zcxe9jfS6pPsw5zISJ5HSZ6B1OxGK9cfbanjAd9FQcA/edit

Comment (MaryamZi) [reply examples/the-main-function/the_main_function.out:7]: Oh okay, didn't know of this. But do these make sense as options? If not, should we consider removing the BBE until we can fully support this?

Comment (manuranga) [reply examples/the-main-function/the_main_function.out:7]: @riyafa can we update it with new changes ?

**Review by manuranga — APPROVED** [review]

**Pull request merged**